colossalai/zero/shard_utils/stateful_tensor_mgr.py

```diff
@@ -0,0 +1,70 @@
+import torch
+from colossalai.context.singleton_meta import SingletonMeta
+from colossalai.utils.cuda import get_current_device
+from colossalai.zero.sharded_param.sharded_param import ShardedParamV2
+from colossalai.zero.sharded_param.sharded_tensor import ShardedTensor
+from colossalai.zero.sharded_param.tensorful_state import TensorState
+from colossalai.zero.shard_utils.tensor_utils import colo_model_data_tensor_move_inline, colo_tensor_mem_usage
+from colossalai.utils.memory_utils.utils import colo_cuda_memory_capacity
+from typing import Set
+from colossalai.utils.memory_tracer import MemStatsCollector
+
+
+class StatefulTensorMgr(SingletonMeta):
+    _stateful_tensor_list: Set[ShardedParamV2] = set()
+
+    def register_param(self, param: ShardedParamV2) -> None:
+        for t in param.get_payload_tensors():
+            assert isinstance(t, ShardedTensor)
+            self._stateful_tensor_list.add(t)
```

Comment:
You add a `tensor` here, but the type hint shows it should be a `ShardedParamV2` instance.